### PR TITLE
Emag wearing and showing

### PR DESCRIPTION
**Pull request by maxymax13**
## About The Pull Request

dreamseeker_Ek3U8n3Cn7
dreamseeker_dieSalU6dw

Lets emags, fake emags (seccies won't be able to tell!), and doorjacks be worn in the ID slot.
Additionally, you can show them off by using them like you would a regular ID.

## Why It's Good For The Game
Makes the items that are distinctly modified ID cards act more like ID cards.
Also adds new ~~fun RP opportunities~~ innovative ways for assistants to taunt sec with fake(?) traitor items.
How long can you walk around with a fake(?) emag in your ID slot before a seccie shift clicks you and chases you down?

## Changelog
:cl:
add: The Syndicate's tech engineers have minorly modified the wiring on their sequencers to not obstruct the ID card clip on jumpsuits, letting you tell the whole station how much of a traitor you are.
add: You can also show them off like you can regular cards, in case the space bar asks for ID.
/:cl:

### CODE REVIEW

File: code/game/objects/items/emags.dm [hunk 1]
```diff
@@ -34,9 +41,16 @@
 	name = "cryptographic sequencer"
 	icon_state = "emag"
 	inhand_icon_state = "card-id"
+	slot_flags = ITEM_SLOT_ID
+	worn_icon_state = "emag"
```
Comment (bobbahbrown) [inline code/game/objects/items/emags.dm:45]:
```suggestion
```

unless im missing something these should be inherited from the parent class

Comment (maxymax13) [reply code/game/objects/items/emags.dm:45]: look closely, it's /emagfake not /emag/fake

Comment (bobbahbrown) [reply code/game/objects/items/emags.dm:45]: oh sorry, i got pranked by github

image

i thought for some reason you were changing ``/obj/item/card/emag/bluespace``

Comment (maxymax13) [reply code/game/objects/items/emags.dm:45]: fair enough, that does look very hard to see in dark mode

**Comment by Tlaltecuhtli** [comment]
make them emag act on bump

**Comment by maxymax13** [comment]
> 
> 
> make them emag act on bump

i'd love to, but the airlock code is kinda spread all over the place. i'll see what i can do

**Comment by Mothblocks** [comment]
dont listen to tlal

**Review by Twaticus — APPROVED** [review]

File: code/game/objects/items/emags.dm [hunk 2]
```diff
@@ -34,9 +41,16 @@
...
 	lefthand_file = 'icons/mob/inhands/equipment/idcards_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/idcards_righthand.dmi'
 
+/obj/item/card/emagfake/attack_self(mob/user) //for assistants with balls of plasteel
+	if(Adjacent(user))
+		user.visible_message("<span class='notice'>[user] shows you: [icon2html(src, viewers(user))] [src.name].</span>", "<span class='notice'>You show \the [src.name].</span>")
```
Comment (Mothblocks) [inline code/game/objects/items/emags.dm:51]:
```suggestion
		user.visible_message("<span class='notice'>[user] shows you: [icon2html(src, viewers(user))] [name].</span>", "<span class='notice'>You show [src].</span>")
```

File: code/game/objects/items/emags.dm [hunk 3]
```diff
@@ -15,9 +15,16 @@
...
 	righthand_file = 'icons/mob/inhands/equipment/idcards_righthand.dmi'
 	item_flags = NO_MAT_REDEMPTION | NOBLUDGEON
+	slot_flags = ITEM_SLOT_ID
+	worn_icon_state = "emag"
 	var/prox_check = TRUE //If the emag requires you to be in range
 	var/type_blacklist //List of types that require a specialized emag
 
+/obj/item/card/emag/attack_self(mob/user) //for traitors with balls of plastitanium
+	if(Adjacent(user))
+		user.visible_message("<span class='notice'>[user] shows you: [icon2html(src, viewers(user))] [src.name].</span>", "<span class='notice'>You show \the [src.name].</span>")
```
Comment (Mothblocks) [inline code/game/objects/items/emags.dm:25]:
```suggestion
		user.visible_message("<span class='notice'>[user] shows you: [icon2html(src, viewers(user))] [name].</span>", "<span class='notice'>You show [src].</span>")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
`\the` is implied, and you should not use `src`.

**Comment by maxymax13** [comment]
i forgot that id card show code did that \the thing because of their funny names, thanks

**Comment by maxymax13** [comment]
what the fuck is "gun go bang"

**Comment by Mothblocks** [comment]
not your problem, broken test that i fixed

**Pull request merged**